packages/shared-types/src/CommonTypes.ts

```diff
@@ -39,3 +39,10 @@ export type UIElement =
  * screen reader
  */
 export type LiveRegion = (() => Element) | Element[] | Element
+
+interface InstUIBaseComponent {
+  componentId: string
+  allowedProps: string[]
+}
+export interface InstUIComponent extends React.ComponentClass<any, any>, InstUIBaseComponent { }
```

Comment:
Hmm, I added it, but now I'm not even sure this is a correct thing to do, since `decorator` expects a plain `React.ComponentClass` which won't have these `componentId` and `allowedProps` static variables on them, so we have to cast every time we use this inside on our decorator implementations.

It would be nice to somehow tell the users with typings that they *need* to have these specific static variables on their components in order to use `withStyle, withDeterministicId`.

Reply:
Isn't the issue simply that the extra props are all required? If I set them as optional, I don't get the errors, because then it is compatible with ComponentClass.

Reply:
Yeah, changing them to be optional solves the issue with the compiler, but I have a feeling this is just a workaround for now.
Maybe our component should extend the `InstUIComponent` interface instead of `React.Component`, but then again the decorators must be general enough to accept any kind of `React.ComponentClass` so I'm not sure to be honest. :D
Anyways, changed the properties to be optional.